### ISSUE TITLE
PROCESS CHANGE: Make LinkedIn posts for new contributors first contributions

### ISSUE THREAD

**Issue by Cali0707** [knative/community#1473]
Cillium currently creates a new LinkedIn post whenever a new contributor has their first contribution merged (example: https://www.linkedin.com/posts/cilium_new-cilium-contributor-the-newest-activity-7131279867601395712-NXy7?utm_source=share&utm_medium=member_desktop)

I think a great way of encouraging new contributors in Knative would be if we also gave them a shoutout on LinkedIn (and maybe one of the slack channels on the CNCF slack) when their first contribution is merged.

The posts would be the responsibility of whoever is managing the LinkedIn account, although @mmejia02 from the UX WG has volunteered to provide some graphics for the post if there is interest. In terms of when we would know someone had made their first contribution, I'm not 100% sure currently. Maybe WG leads would have to reach out when they see someone's first contribution to e.g. eventing?

cc @aliok

**Comment by aliok** [comment]
I think this would be great.

Things needed would be:
1. How to recognize someone's first contribution? We might need some tooling for that because the people who can post on social media wouldn't be able to track who has made their first contribution.
2. We need to set up a process to send such posts for @knative/steering-committee with details like
  - What to post (text content, what kind of details, graphics (thanks @mmejia02 for offering help))
  - When to post
  - Who should post

(1) -->  could you reach out to Cilium to learn more about how they do this @Cali0707 ?
(2) --> I can try to setup a process at SC side after we know more about (1) and how Cilium does this.

Does this sound good?

Related: https://github.com/cncf/tag-contributor-strategy/issues/50

**Comment by Cali0707** [comment]
> could you reach out to Cilium to learn more about how they do this @Cali0707 ?

Yes, I'll reach out and ping you here when I have a response @aliok :+1:

**Comment by aliok** [comment]
@xmulligan or @Carla-Gaggini might help with it @Cali0707

**Comment by xmulligan** [comment]
Sure, happy to comment here. We have [Orbit](https://orbit.love/) and it notifies you every time there is a new contributor. We usually batch schedule them every few weeks rather than as they come in.

I think you should also be able to pull it from DevStats. Here is the Cilium [one](https://cilium.devstats.cncf.io/d/52/new-contributors-table?orgId=1), though it looks like the knative [one](https://knative.devstats.cncf.io/d/52/new-contributors-table?orgId=1) isn't working right now for some reason.

**Comment by aliok** [comment]
Created a DevStats issue: https://github.com/cncf/devstats/issues/46

About Orbit:
- I like it.
- There's a free subscription: https://orbit.love/pricing with some limitations. Hard to evaluate if we need the features in the paid plans without actually trying*, or asking @xmulligan 


*trying: We need to install a GitHub app on Knative GitHub organizations, so, we need SC/TOC approval for trying it out.

**Comment by xmulligan** [comment]
The notification about new contributors is in the free plan. That's actually about all we use Orbit for 😅

**Comment by aliok** [comment]
> The notification about new contributors is in the free plan. That's actually about all we use Orbit for 😅

:) Thanks @xmulligan .

I know I've asked already a lot from you, but can you take a few screenshots for us next time you're in that tool? That would help us "selling" the idea to Knative steering committee.

**Comment by aliok** [comment]
And another thing, there's a relevant document in TAG Contributor Strategy website: https://contribute.cncf.io/maintainers/community/crm-runbook/

I was reading that doc before, but it is somewhat abstract. If you're interested, it can be improved with more concrete examples. Just another contribution opportunity 😉

**Comment by xmulligan** [comment]
<img width="1502" alt="Screenshot 2023-12-19 at 16 41 31" src="https://github.com/knative/community/assets/29048870/8abc2b47-508e-4e07-8ebd-96bd9adb6e9d">
Sure here is an example. You can see on the right side the notifications when someone merges their first PR.

To be completely honest, this is the single feature we use Orbit for so I may not be the best to update the CRM runbook 😅

**Comment by aliok** [comment]
I was already writing a tool that fetches data from GitHub. I've used that tool to fetch new contributors for the last month (e.g. today=Jan-30-2024, lastMonth=Dec-2023) and add them to a list of contributors.

https://github.com/aliok/knative-contrib-report/blob/main/250-build-contributor-list/contributors.json

The list is built by periodic GitHub actions executions.

**Comment by aliok** [comment]
Well, the code currently is not reviewed. If a second pair of eyes can do a review, and a sanity check, then we can use it :)

**Comment by aliok** [comment]
Orbit needs to be installed to Knative GitHub organization to work.

It needs readonly permissions.

This needs to be voted by the SC.

Although I believe my house-made solution at https://github.com/knative/community/issues/1473#issuecomment-1917269164 works fine, it might be nicer to see other aspects as well. Also, it can be used to verify my house-made solution :)

I am also not sure if a free account is enough to get the first contributors for `knative` and `knative-extensions` GitHub organizations.


<img width="788" alt="Screenshot 2024-02-04 at 00 12 01" src="https://github.com/knative/community/assets/376732/ae93f2db-d57f-455f-81d2-c35275f964c2">

**Comment by Leo6Leo** [comment]
I love this! But I'd like to suggest an approach that focuses on celebrating contributions once individuals have made a significant impact, or around 3-5 contributions, to the community. Here’s why :

1. This encourages a culture of sustained quality and effort. It makes sure that those who invest a lot of time and expertise feel genuinely valued, and differentiate between simple tasks and significant contributions. 

2. prevents the 1-time Contributions: We hope to create genuine engagement instead of one-off contributions made solely for recognition. This ensures that people are not just contributing for the sake of getting recognition on LinkedIn.

**Comment by aliok** [comment]
@Leo6Leo Thanks a lot for thinking this through. Makes sense.

We might take a different approach for sure.

Maybe we need to separate this into 2 things:
- Celebrating the first contributions of people (batch celebration)
  - We have the tooling for this already (see https://github.com/knative/community/issues/1473#issuecomment-1925457316)
- Celebrating the significant impact of people (individual celebration)
  - We have the list of people who made their first contributions, and when they made it
  - Every month, we can check the contributors who made their first contribution 3-4 months ago and check if they've continued since then.
  - We might need some tooling for the 2nd part.

Great input!

**Comment by aliok** [comment]
I've collected the "first contributors" of January 2024 and I wanted to send a celebration message on Slack: https://cloud-native.slack.com/archives/C04LGHDR9K7/p1706996186936599

**Comment by aliok** [comment]
Based on above, I made a post from my own account: https://www.linkedin.com/posts/aliok_knative-opensource-community-activity-7161640368281710592-D6Tp?utm_source=share&utm_medium=member_desktop

I think, current approach will work for the first contributions.

For the second group (impactful contributors), we need to do some additional work.

So, the outcome of this issue should be:
- Guideline for checking who made their first contributions, how to collect their information, what kind of post to send, etc.
- A 2nd guideline for the 2nd group of contributors (need an experiment like I did though)

**Comment by aliok** [comment]
@Cali0707 @Leo6Leo and I are doing this celebration posts regularly now (every month). However, we need to document the process and give this task over to the to-be-defined social media manager.
